src/ctl/cart_ctl.c

```diff
@@ -377,18 +383,69 @@ parse_args(int argc, char **argv)
 	return rc;
 }
 
+void
+print_uri_cache(struct crt_ctl_get_uri_cache_out *out_uri_cache_args)
+{
+	char					*ptr;
+	char					*uri;
+	size_t					 buf_len;
+	d_rank_t				 rank;
+	int					 tag;
+	int					 len;
+	int					 size = 0;
+	int					 uri_size;
+
+	ptr = out_uri_cache_args->cguc_grp_info.iov_buf;
+
+	buf_len = out_uri_cache_args->cguc_grp_info.iov_buf_len;
+
+	size = 0;
+
+	while (size < buf_len) {
```

Comment:
Oh, I see what's happening here.  You should define the RPC with a CRT_ARRAY in the output, and have CaRT do the packing, rather than using a iovec.

Reply:
> Oh, I see what's happening here. You should define the RPC with a CRT_ARRAY in the output, and have CaRT do the packing, rather than using a iovec.

I've never used crt_array in an RPC. Looks like you are right, cart can pack crt_array automatically.